src/constraint_system/ecc/scalar_mul/fixed_base/mod.rs

```diff
@@ -0,0 +1,326 @@
+use crate::constraint_system::ecc::curve_addition::fixed_base_gate::WnafRound;
+use crate::constraint_system::ecc::{Point, PointScalar};
+use crate::constraint_system::{variable::Variable, StandardComposer};
+use dusk_bls12_381::Scalar as BlsScalar;
+use dusk_jubjub::{AffinePoint, ExtendedPoint, Fr as JubJubScalar};
+
+fn compute_wnaf_point_multiples(generator: ExtendedPoint, num_bits: usize) -> Vec<AffinePoint> {
+    assert!(generator.is_prime_order().unwrap_u8() == 1);
+
+    let mut multiples = vec![ExtendedPoint::default(); num_bits];
+    multiples[0] = generator;
+    for i in 1..num_bits {
```

Comment:
Consider using iterators instead

Reply:
In this situation, I believe the for-loop is correct. @CPerezz Can you describe the change necessary to use iterators?